Dependencies/napi/napi-direct/source/env.cc

```diff
@@ -1,11 +1,18 @@
 #include <napi/env.h>
+#include <sstream>
 
 namespace Napi
 {
-    Napi::Value Eval(Napi::Env env, const char* string, const char* sourceUrl)
+    Napi::Value Eval(Napi::Env env, const char* source, const char* sourceUrl)
     {
         napi_value result;
-        NAPI_THROW_IF_FAILED(env, napi_run_script(env, Napi::String::New(env, string), sourceUrl, &result));
+
+        // Append the source URL so V8 can locate the file.
+        std::ostringstream sourceWithURL;
+        sourceWithURL << source << "\n";
```

Comment:
should this always be done or should it only be used for v8 scenarios behind a macro?

Reply:
This code should probably be inside the napi_run_script function for V8.

Reply:
Actually, does V8 have a way to specify this without changing the string?

Reply:
I scoured the v8 docs looking for a way to pass the source url to the engine and there doesn't seem to be a way. Fortunately at least the inspector seems to identify and hide the string from the source so it's basically invisible to the user.